--- ./node_modules/next/dist/server/render.js
+++ ./node_modules/next/dist/server/render.js
@@ -406,7 +406,7 @@
         if (data == null) {
             throw new Error(_constants.GSP_NO_RETURNED_VALUE);
         }
-        const invalidKeys = Object.keys(data).filter((key)=>key !== "revalidate" && key !== "props" && key !== "redirect" && key !== "notFound");
+        const invalidKeys = Object.keys(data).filter((key)=>key !== "revalidate" && key !== "props" && key !== "redirect" && key !== "notFound" && key !== "expires");
         if (invalidKeys.includes("unstable_revalidate")) {
             throw new Error(_constants.UNSTABLE_REVALIDATE_RENAME_ERROR);
         }
@@ -468,6 +468,7 @@
         }
         props1.pageProps = Object.assign({}, props1.pageProps, "props" in data ? data.props : undefined);
         renderOpts.revalidate = "revalidate" in data ? data.revalidate : undefined;
+        renderOpts.expires = "expires" in data ? data.expires : undefined;
         renderOpts.pageData = props1;
         // this must come after revalidate is added to renderOpts
         if (renderOpts.isNotFound) {
