--- ../node_modules/next/dist/server/send-payload/revalidate-headers.js
+++ ../node_modules/next/dist/server/send-payload/revalidate-headers.js
@@ -12,9 +12,9 @@
         if (options.revalidate < 1) {
             throw new Error(`invariant: invalid Cache-Control duration provided: ${options.revalidate} < 1`);
         }
-        res.setHeader("Cache-Control", `s-maxage=${options.revalidate}, stale-while-revalidate`);
+        res.setHeader("Expires", options.expires && new Date(Date.now() + options.revalidate * 1000).toUTCString());
     } else if (options.revalidate === false) {
-        res.setHeader("Cache-Control", `s-maxage=31536000, stale-while-revalidate`);
+        res.setHeader("Cache-Control", `s-maxage=31536000`);
     }
 }
 
